cli/flox/src/commands/list.rs

```diff
@@ -115,22 +121,40 @@ impl List {
     /// e.g. `pip: python3Packages.pip (20.3.4)`
     ///
     /// This is the default mode
-    fn print_extended(mut out: impl Write, packages: &[PackageToList]) -> Result<()> {
+    fn print_extended(
+        mut out: impl Write,
+        packages: &[PackageToList],
+        upgrades: Option<SingleSystemUpgradeDiff>,
+    ) -> Result<()> {
         for p in packages {
+            let install_id = match p {
+                PackageToList::Catalog(_, p) => &p.install_id,
+                PackageToList::Flake(_, p) => &p.install_id,
+                PackageToList::StorePath(p) => &p.install_id,
+            };
+            let upgrade_available = if upgrades
+                .as_ref()
+                .is_some_and(|diff| diff.contains_key(install_id))
+            {
+                " - upgrade available"
+            } else {
+                ""
+            };
+
             match p {
-                PackageToList::Catalog(p) => {
+                PackageToList::Catalog(_, p) => {
                     writeln!(
                         &mut out,
-                        "{id}: {path} ({version})",
+                        "{id}: {path} ({version}{upgrade_available})",
                         id = p.install_id,
-                        path = p.rel_path,
-                        version = p.info.version.as_deref().unwrap_or("N/A")
+                        path = p.attr_path,
+                        version = p.version,
```

Comment:
question: is version truly required by the catalog, or does it ever get set to `""`?

Reply:
We were just setting version to `Some(pkg.version)` before anyways, so this isn't a regression. And from the test database it doesn't look like there are any empty versions